OneLogin How-to (#1838)
spectrocloud/librarium: docs/docs-content/user-management/saml-sso/palette-sso-with-onelogin.md

  ![Add Security Policy](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_group-sec-policy.png)

15. You will automate the mapping of a user to a role and group by creating a *Mapping*. Navigate to **Users** --> **Mappings** and select **New Mapping**. Assign the mapping a name and set it to map every member of the Administrators group to assign the admin role automatically. Set the **MemberOf** value to **Administrators**. The latter step is essential, so the response from OneLogin contains the group name, which you will match with a Team in Palette. Without explicitly setting the **MemberOf** value, Palette will not get the group name and will not be able to set the correct RBAC settings for your user. 

  ![Add Mapping](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_mapping-role-group.png)

16. Navigate to **Users** --> **Users** and select your user. Select the **Authentication** tab and select the group and security policy you created earlier. Click on **Save**. 

  ![Add User to Group](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_user-auth-group.png)

### Enable OIDC in Palette 

> 17. Navigate back to Palette and fill in the **Client ID**, **Client Secret**, and **Issuer URL** values. Add the **groups** scope to the **Scopes** field. Without the proper scopes, Palette won't receive the group name. Click **Enable** to continue. 

Reviewer: ```suggestion
17. Navigate back to Palette and fill in the **Client ID**, **Client Secret**, and **Issuer URL** values. 

18. Next, add the **groups** scope in the **Scopes** field, and click **Enable** to continue. 

:::caution
The proper scope is required for Palette to receive the group name. 
:::
```

  ![Full OIDC config](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_oidc-full-palette.png)

### Create Teams in Palette

> 

Reviewer: Start with 1. in this new section and provide an introduction. You can borrow from line 117. For example:

"You must create a team in Palette so that Palette can map the team with groups configured in OneLogin. The team name must match the value returned for the groups configured in OneLogin. In this example, you will create an **Administrators** team in Palette. This team name matches the Administrators value assigned to each member of the Administrators group in OneLogin."

18. In order for Palette to map the groups configured in OneLogin, you need to create a team in Palette. The team name needs to match the value that is returned for the groups configured in OneLogin. In this example, you will create an **Administrators** team in Palette. This team name also matches with the Administrators value that is assigned to each member of the Administrators group in OneLogin. 

>   In Palette, navigate to left **Main Menu**, select **Tenant Settings** and click on **Users & Teams**. Next, select **Teams**, and click on **Create team**. You only need to fill in the team name. Don't manually add members, as OneLogin will manage the users. In OneLogin, users who are part of the mapped group, administrators, will be automatically added to the team. Click **Confirm** to continue. 

Reviewer: ```suggestion
  1. From the left **Main Menu** in Palette, click on **Users & Teams**. 
  
  2. Select the **Teams** tab, and click on the **Create team** button. 
  
  3. Provide *only* the team name and ignore the **Members** field, as OneLogin will manage the users. In OneLogin, users who are part of the mapped "Administrators" group will be automatically added to the team. Click **Confirm** to continue. 
```

  ![Create New Team](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_new-team.png)

> 19. Next, you need to assign the members of this team a set of permissions. Assign all members the **Tenant Admin** permissions. You can customize the assigned permissions as needed, but for this example admin access is granted. To set the correct permissions, navigate to left **Main Menu**, select **Tenant Settings** --> **Users & Teams** --> **Teams**. Select the team you created. Click on **Tenant Roles** --> **Add Tenant Role** and select the **Tenant Admin** box. This will auto-select all other permission boxes.  Click **Confirm**.

Reviewer: Step needs to be broken out into individual steps.

Avoid menu cascade (-->).

## Validate

Use the following steps to validate the configuration.

1. Log out of Palette by clicking on the user **drop-down Menu** in the top-right corner and selecting **Log Out**. You will be redirected to a new login screen.

  ![Logout](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_logout.png)

2. The login screen now defaults to a **Sign In** button. Click on **Sign In**. You will be required to authenticate with OneLogin. If you are already authenticated with OneLogin you will be automatically signed in to Palette with the proper permissions inherited from the respective Palette team you are a member of. If you cannot sign in, you can use the username and password method. 

  ![SSO Login Screen](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_login-screen.png)

> 3. You can check what teams you are mapped to by navigating to the left **Main Menu** and selecting **Tenant Settings**. Select **Users & Teams** --> **Users**. Find your user name to review the teams you are a member of. 

Reviewer: ```suggestion
3. To check which teams you are mapped to, navigate to the left **Main Menu** and select **Users & Teams**. In the **Users** tab, find your user name to review the teams you are a member of. 
```

Reviewer: You might check the navigation throughout the doc. Users & Teams is in the Main Menu.

  ![Check Team Member](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_team-member.png)

:::tip

> With the [OpenID Connect Inspector](https://developers.onelogin.com/openid-connect/inspector), you can send requests to OneLogin and check what is sent in the payload. This way you can make sure that you are using the right claims and scopes. When using this, you will need to add the Inspector callback URL to your applications Redirect URIs list. Check out the - [OpenID Connect Inspector Tutorial](https://youtu.be/do0agd71hE8) to learn more.

Reviewer: ```suggestion
With the [OpenID Connect Inspector](https://developers.onelogin.com/openid-connect/inspector), you can send requests to OneLogin and check what is sent in the payload. This way you can ensure you are using the correct claims and scopes. When using this, you must add the Inspector callback URL to your application's Redirect URIs list. Check out the - [OpenID Connect Inspector Tutorial](https://youtu.be/do0agd71hE8) to learn more.
```

Reviewer: "When using this ... " is vague. What does "this" refer to?

![Add Security Policy](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_group-sec-policy.png)

19. You will automate the mapping of a user to a role and group by creating a *Mapping*. Navigate to **Users**, followed by **Mappings** and select **New Mapping**. 

20. Assign the mapping a name and set it to map every member of the Administrators group to assign the admin role automatically. Set the **MemberOf** value to **Administrators**. The latter step is essential, so the response from OneLogin contains the group name, which you will match with a Team in Palette. Without explicitly setting the **MemberOf** value, Palette will not get the group name and will not be able to set the correct RBAC settings for your user. 

  ![Add Mapping](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_mapping-role-group.png)

21. Navigate to **Users** screen and select your user. 

22. Select the **Authentication** tab and select the group and security policy you created earlier. Click on **Save**. 

  ![Add User to Group](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_user-auth-group.png)

### Enable OIDC in Palette 

> 23. Navigate back to Palette, and from the **left Main Menu**, select **Tenant Settings**. Next, select **SSO** and click on the **OIDC** tab.

Reviewer: ```suggestion
23. Navigate back to Palette, and from the left **Main Menu**, select **Tenant Settings**. Next, select **SSO** and click on the **OIDC** tab.
```

24. Fill in the fields **Client ID**, **Client Secret**, and **Issuer URL** with the values you were provided by OneLogin. 

25. Next, add the **groups** scope in the **Scopes** field, and click **Enable** to continue. 

  :::caution
  Ensure the expected scopes are added. Otherwise, Palette may be unable to retrieve the group name. 
  :::

  ![Full OIDC config](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_oidc-full-palette.png)

You now have a working configuration for OneLogin as a third-party IdP in Palette. Check out the [Create Teams in Palette](#create-teams-in-palette) section to learn how to create teams in Palette and map them to groups in OneLogin.

## Validate

Use the following steps to validate the configuration.

> 1. Log out of Palette by clicking on the user **drop-down Menu** in the top-right corner and selecting **Log Out**. You will be redirected to a new login screen.

Reviewer: ```suggestion
1. Log out of Palette by navigating to the top right **User Menu** and selecting **Log Out**. You will be redirected to a new login screen.
```

  ![Logout](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_logout.png)

> 2. The login screen now defaults to a **Sign In** button. Click on **Sign In**. You will be required to authenticate with OneLogin. If you are already authenticated with OneLogin you will be automatically signed in to Palette with the proper permissions inherited from the respective Palette team you are a member of. If you cannot sign in, you can use the username and password method. 

Reviewer: ```suggestion
2. In the login screen that displays, click the **Sign In** button. You will be required to authenticate with OneLogin. If you are already authenticated with OneLogin, you will be signed in automatically to Palette with the proper permissions inherited from the Palette team you are a member of. If you cannot sign in, you can use the username and password method. 
```

:::tip

With the [OpenID Connect Inspector](https://developers.onelogin.com/openid-connect/inspector), you can send requests to OneLogin and check what is sent in the payload. This way, you can ensure you are using the correct claims and scopes. Add the Inspector's callback URL to your application's Redirect URIs list when using the Inspector. Check out the - [OpenID Connect Inspector Tutorial](https://youtu.be/do0agd71hE8) to learn more.

:::info

## Create Teams in Palette

You must create a team in Palette so that Palette can map the team with groups configured in OneLogin. The team name must match the value returned for the groups configured in OneLogin. In this example, you will create an Administrators team in Palette. This team name matches the Administrators value assigned to each member of the Administrators group in OneLogin.

### Prerequisites

- An active Palette account with administrative permissions.

- A OneLogin mapping that maps the user to a group.

### Team Creation

1. Log in to [Palette](https://console.spectrocloud.com/) as a tenant administrator.

> 2. From the **left Main Menu**, select **Users & Teams**.

Reviewer: ```suggestion
2. From the left **Main Menu**, select **Users & Teams**.
```

3. Next, select **Teams**, and click on **Create team**. 

> 4. You only need to fill in the team name. Don't manually add members, as OneLogin will manage the users. In OneLogin, users who are part of the mapped group, administrators, will be automatically added to the team. Click **Confirm** to continue. 

Reviewer: ```suggestion
4. You only need to fill in the team name. Do not manually add members, as OneLogin will manage the users. In OneLogin, users who are part of the mapped administrators group will be automatically added to the team. Click **Confirm** to continue. 
```

  ![Create New Team](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_new-team.png)

> 5. Next, you need to assign the members of this team a set of permissions. Assign all members the **Tenant Admin** permissions. You can customize the assigned permissions as needed, but for this example admin access is granted. 

Reviewer: ```suggestion
5. Next, you need to assign the members of this team a set of permissions. Assign all members the **Tenant Admin** permissions. You can customize the assigned permissions as needed. For this example admin access is granted. 
```

> 6. To set the correct permissions, select the team you created. Click on the **Tenant Roles** tab.

Reviewer: ```suggestion
6. To set the correct permissions, select the team you created, and click on the **Tenant Roles** tab.
```

7. Next, click on **Add Tenant Role** and select the **Tenant Admin** box. This will automatically select all other permission boxes. Click on **Confirm**.

8. You should have a configuration similar to the following image. 

  ![Permissions](/oidc-onelogin-images/user-management_saml-sso_palette_sso_with_onelogin_roles-full.png)

You have now configured Palette to use OneLogin as a third-party IDP. Use the above steps to create additional groups in OneLogin and Palette. 

> As you start deploying clusters, keep in mind that if you use Palette's [Extended Kubernetes distribution](../../integrations/kubernetes.md) (PXK), you can configure the cluster to inherit OIDC configurations from the tenant. Refer to the [Palette eXtended Kubernetes](../../integrations/kubernetes.md#configure-custom-oidc) reference page to learn more about out-of-the-box support for OIDC.

Reviewer: ```suggestion
As you start deploying clusters, keep in mind that if you use the [Palette eXtended Kubernetes (PKE)](../../integrations/kubernetes.md) distribution, you can configure the cluster to inherit OIDC configurations from the tenant. Refer to the [Palette eXtended Kubernetes](../../integrations/kubernetes.md#configure-custom-oidc) reference page to learn more about out-of-the-box support for OIDC.
```

> ### Validate

Reviewer: ```suggestion
## Validate
```

> To verify that the team is created and the correct permissions are assigned, log in to Palette as a user who is a member of the team.

Reviewer: ```suggestion
To verify the team is created and the correct permissions are assigned, log in to Palette as a user who is a member of the team.
```

1. Log in to [Palette](https://console.spectrocloud.com/) as a user who is a member of the team you created and available in OneLogin.

2. Authenticate with OneLogin once you are redirected to the login screen. A successful login will redirect you to the Palette dashboard and indicate that you are logged in as a member of the team you created. 

> 3. Attempt to access the **Tenant Settings** page to ensure you have the correct permissions. From the left **Main Menu**, select the scope, **Tenant Admin**. Next, select **Tenant Settings**. If you have the correct permissions, you will be able to access the settings page. 

Reviewer: ```suggestion
3. Ensure you can access the **Tenant Settings** page. Select the **Tenant Admin** scope. From the left **Main Menu**, select **Tenant Settings**. If you have the correct permissions, you will be able to access the settings page. 
```
